crates/topos-tce-api/src/stream/mod.rs

```diff
@@ -150,14 +150,38 @@ impl Stream {
                     }
                 }
 
-                Some(_stream_packet) = self.inbound_stream.next() => {
-
+                // We currently open the stream, but no other message from the client is getting processed.
+                // We are using this open connection to communicate `delivered_certificates` to the client.
+                Some(stream_packet) = self.inbound_stream.next() => {
+                    match stream_packet {
+                        Ok((_request_id, _message)) => {
+                            trace!("Received message from stream_id: {:?}", self.stream_id);
+                        }
+                        Err(error) => {
+                            match error.kind {
+                                StreamErrorKind::StreamClosed => {
+                                    // Currently, when the client is closing the connection,
+                                    // we don't receive a StreamClosed error. Further investigation is needed.
```

Comment:
`StreamClosed` error are generated when the server detect a problem with either the connection or the protocol used and thus close the connection.